File: rb/features.rb
# IinLookup SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module IinLookupFeatures
  def self.make_feature(name)
    case name
    when "base"
      IinLookupBaseFeature.new
    when "test"
      IinLookupTestFeature.new
    else
      IinLookupBaseFeature.new
    end
  end
end

File: py/iinlookup_sdk/features.py
# IinLookup SDK feature factory

from iinlookup_sdk.feature.base_feature import IinLookupBaseFeature
from iinlookup_sdk.feature.test_feature import IinLookupTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IinLookupBaseFeature(),
        "test": lambda: IinLookupTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
